docs/shared/concepts/typescript-project-linking.md

```diff
@@ -0,0 +1,277 @@
+# Typescript Project Linking
+
+The naive way to reference code in a separate project is to use a relative path in the `import` statement.
+
+```ts
+import { someFunction } from '../../teamA/otherProject';
+
+const result = someFunction();
+```
+
+The problem with this approach is that all your import statements become tied to your folder structure. Developers need to know the full path to any project from which they want to import code. Also, if `otherProject` ever moves locations, there will be superfluous code changes across the entire repository.
+
+A more ergonomic solution is to reference your local projects as if they were external npm packages and then use a project linking mechanism to automatically resolve the project file path behind the scenes.
+
+```ts
+import { someFunction } from '@myorg/otherProject';
+
+const result = someFunction();
+```
+
+There are two different methods that Nx supports for linking TypeScript projects: package manager workspaces and TypeScript path aliases. Project linking with TS path aliases was available with Nx before package managers offered a workspaces project linking approach. The Nx Team has since added full support for workspaces because (1) it has become more common across the TypeScript ecosystem and (2) using workspaces allows Nx to also enable TypeScript project references for improved build performance.
+
+## Project Linking with Workspaces
+
+To create a new Nx workspace that links projects with package manager workspaces, use the `--workspaces` flag.
+
+```shell
+npx create-nx-workspace --workspaces
+```
+
+### Set Up Package Manager Workspaces
+
+The configuration for package manager workspaces varies based on which package manager you're using.
```

Comment:
I know you mention the package.json linking in the various tabs since PNPM has a different protocol. Can we bring it up more clearly though that for the "workspaces" based pacakge linking you need to have the dependency to the packages defined in the project's `package.json`?

like if `A -> B` then `A`'s `package.json` needs to have a dependency to `B`. 

I know in theory NPM (and maybe yarn and bun) don't strictly require this, while PNPM (and yarn pnp?) do. But I'd almost lean towards defining it so things don't go wrong.